Error in generated code for `self.position().offset()`

I would expect the following to generate code printing the offset of the unit field, but it looks like invalid code is generated so this does not compile

```ruby
module foo;

public type X = unit {
    : uint8 { print self.position().offset(); }
};
```

```console
$ spicyc -j foo.spicy
/private/var/folders/ht/vtb0rkbd4ws15vx02xdb84zw0000gn/T/foo_4401a488ba3fc7e5-8f847f58a5b1b22b.cc:322:44: error: no member named 'offset' in 'std::optional<hilti::rt::stream::SafeConstIterator>'; did you mean to use '->' instead of '.'?
    ::hilti::rt::print((*__self).__position.offset(), ::hilti::rt::Bool(true));
                                           ^
                                           ->
1 error generated.
[error] spicy-driver: JIT compilation failed
```

--- a/spicy/toolchain/src/compiler/codegen/codegen.cc
+++ b/spicy/toolchain/src/compiler/codegen/codegen.cc
@@ -201,7 +201,7 @@ struct VisitorPass2 : public hilti::visitor::PreOrder<void, VisitorPass2> {
     }
 
     result_t operator()(const operator_::unit::Position& n, position_t p) {
-        replaceNode(&p, builder::member(n.op0(), ID("__position")));
+        replaceNode(&p, builder::deref(builder::member(n.op0(), ID("__position"))));
     }
 
     result_t operator()(const operator_::unit::Input& n, position_t p) {

--- a/tests/spicy/types/unit/offset.spicy
+++ b/tests/spicy/types/unit/offset.spicy
@@ -6,13 +6,22 @@ module Mini;
 import spicy;
 
 type Sub = unit {
-    on %init   { print "Sub:init", self.offset(), self.position(); }
-    x: uint16  { print "Sub:x", self.offset(), self.position(); }
-    y: uint8   { print "Sub:y", self.offset(), self.position(); }
+    on %init { print "Sub:init", self.offset(), self.position(); }
+    x: uint16 { print "Sub:x", self.offset(), self.position(); }
+    y: uint8 { print "Sub:y", self.offset(), self.position(); }
 };
 
 public type Test = unit {
-    on %init           { print "Main:init", self.offset(), self.position(); }
-    : Sub[3]   foreach { print "Main:Sub", self.offset(), self.position(); }
-    on %done           { print "Main:done", self.offset(), self.position(); }
+    on %init {
+        print "Main:init", self.offset(), self.position();
+        assert self.position().offset() == self.offset();
+    }
+    : Sub[3] foreach {
+        print "Main:Sub", self.offset(), self.position();
+        assert self.position().offset() == self.offset();
+    }
+    on %done {
+        print "Main:done", self.offset(), self.position();
+        assert self.position().offset() == self.offset();
+    }
 };

